config/routes.rb

```diff
@@ -28,6 +28,9 @@
   end
 
   get '/changes/page/:change_id' => 'changes#page', as: :page_change
+  get '/changes/page/:change_id/html' => 'changes#html', as: :show_html
+  get '/changes/page/:change_id/html/raw' => 'changes#raw_html', as: :raw_html
+  get '/changes/page/:change_id/html/download' => 'changes#download', as: :download_html
```

Comment:
Going off the existing pattern here these urls should be
```
/changes/html/:change_id
/changes/raw_html/:change_id
/changes/download_html/:change_id
```
I think also the `as:` helper names should include the name of the model they're associated with so `:show_change_html` for example.

Reply:
So if I move this to PageSnapshotController, should the route also change? 
It feels like since we're accessing these through the /changes/page/ urls it makes sense to keep them together here. Is that OK?

Reply:
The URL should change to match the controller. Maybe:
```
/page_snapshots/:page_snapshot_id
/page_snapshots/:page_snapshot_id/raw
/page_snapshots/:page_snapshot_id/download
```
The way to think about URLs is ultimately as a "resource locator" so they should reflect the resource that they point to. These are pointing to page snapshots in various formats. If you wanted to make the URL subordinate to the page resource that could work too. I could see something like
```
/page/:page_id/snapshots/:snapshot_id
...
```
also making sense. I don't think it makes a ton of sense to link them to changes though. I think of changes as subordinate to snapshots, not the other way around.